### PR TITLE
Add functional tests

### PR DESCRIPTION
**Pull request by lecorguille**
I forgot to rename script in the different wrappers

### CODE REVIEW

**Comment by lecorguille** [comment]
So here is my brand new PR
Although, it works when I use `planemo serve`, unfortunatly, I always get for both denovo_map and ref_map, the same error when I use `planemo test`.

The part of trace log:
```
[...]
STACKS : Reference map ( stacks_refmap ) > Test-1 ... requests.packages.urllib3.connectionpool INFO 2016-03-08 22:46:39,633 Starting new HTTP connection (1): localhost
galaxy.web.framework.webapp INFO 2016-03-08 22:46:39,637 Session authenticated using Galaxy master api key
requests.packages.urllib3.connectionpool DEBUG 2016-03-08 22:46:39,645 "GET /api/users?key=test_key HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool INFO 2016-03-08 22:46:39,657 Starting new HTTP connection (1): localhost
galaxy.web.framework.webapp INFO 2016-03-08 22:46:39,661 Session authenticated using Galaxy master api key
requests.packages.urllib3.connectionpool DEBUG 2016-03-08 22:46:39,951 "POST /api/users HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool INFO 2016-03-08 22:46:39,953 Starting new HTTP connection (1): localhost
galaxy.web.framework.webapp INFO 2016-03-08 22:46:39,957 Session authenticated using Galaxy master api key
requests.packages.urllib3.connectionpool DEBUG 2016-03-08 22:46:39,975 "POST /api/users/2891970512fa2d5a/api_key HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool INFO 2016-03-08 22:46:39,977 Starting new HTTP connection (1): localhost
requests.packages.urllib3.connectionpool DEBUG 2016-03-08 22:46:40,076 "POST /api/histories HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool INFO 2016-03-08 22:46:40,077 Starting new HTTP connection (1): localhost
galaxy.tools DEBUG 2016-03-08 22:46:40,136 Validated and populated state for tool request (24.271 ms)
requests.packages.urllib3.connectionpool DEBUG 2016-03-08 22:46:40,137 "POST /api/tools HTTP/1.1" 400 None
ERROR

======================================================================
ERROR: STACKS : Reference map ( stacks_refmap ) > Test-1
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/galaxy/test/functional/test_toolbox.py", line 291, in test_tool
    self.do_it( td )
  File "/usr/local/galaxy/test/functional/test_toolbox.py", line 60, in do_it
    raise e
RunToolException: Error creating a job for these tool inputs - Unknown error occurred while processing request.
-------------------- >> begin captured logging << --------------------
requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost
galaxy.web.framework.webapp: INFO: Session authenticated using Galaxy master api key
requests.packages.urllib3.connectionpool: DEBUG: "GET /api/users?key=test_key HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost
galaxy.web.framework.webapp: INFO: Session authenticated using Galaxy master api key
requests.packages.urllib3.connectionpool: DEBUG: "POST /api/users HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost
galaxy.web.framework.webapp: INFO: Session authenticated using Galaxy master api key
requests.packages.urllib3.connectionpool: DEBUG: "POST /api/users/2891970512fa2d5a/api_key HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost
requests.packages.urllib3.connectionpool: DEBUG: "POST /api/histories HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost
galaxy.tools: DEBUG: Validated and populated state for tool request (24.271 ms)
requests.packages.urllib3.connectionpool: DEBUG: "POST /api/tools HTTP/1.1" 400 None
--------------------- >> end captured logging << ---------------------

----------------------------------------------------------------------
XML: /private/var/folders/bm/3wkhsl3j1h747_jm4xthmh980000kd/T/tmpavcgsq/xunit.xml
----------------------------------------------------------------------
Ran 1 test in 0.510s

FAILED (errors=1)
functional_tests.py INFO 2016-03-08 22:46:40,159 Shutting down
functional_tests.py INFO 2016-03-08 22:46:40,159 Shutting down embedded web server
functional_tests.py INFO 2016-03-08 22:46:40,161 Embedded web server stopped
functional_tests.py INFO 2016-03-08 22:46:40,161 Shutting down app
galaxy.jobs.handler INFO 2016-03-08 22:46:40,161 sending stop signal to worker thread
galaxy.jobs.handler INFO 2016-03-08 22:46:40,162 job handler queue stopped
galaxy.jobs.runners INFO 2016-03-08 22:46:40,162 TaskRunner: Sending stop signal to 2 worker threads
galaxy.jobs.runners INFO 2016-03-08 22:46:40,162 LocalRunner: Sending stop signal to 5 worker threads
galaxy.jobs.handler INFO 2016-03-08 22:46:40,163 sending stop signal to worker thread
galaxy.jobs.handler INFO 2016-03-08 22:46:40,164 job handler stop queue stopped
functional_tests.py INFO 2016-03-08 22:46:40,165 Embedded Universe application stopped
functional_tests.py INFO 2016-03-08 22:46:40,165 Cleaning up temporary files in /var/folders/bm/3wkhsl3j1h747_jm4xthmh980000kd/T/tmpRWUwfF/tmp5ljvNS
Exception in thread Thread-3:
Traceback (most recent call last):
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/usr/local/galaxy/.venv/lib/python2.7/site-packages/paste/httpserver.py", line 1101, in serve_forever
    self.handle_request()
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/SocketServer.py", line 276, in handle_request
    fd_sets = _eintr_retry(select.select, [self], [], [], timeout)
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/SocketServer.py", line 155, in _eintr_retry
    return func(*args)
error: (9, 'Bad file descriptor')
[...]
```
I got this log on my Mac. But I also got the first part of the error on a regular linux server (CentOS6) (`RunToolException: Error creating a job for these tool inputs - Unknown error occurred while processing request.`)

Maybe if I find time, I will be able to check piece by piece ... But I guess I will same time, so energy, so CO2 footprint if I submit that to you :smile: 

Cheers

File: tools/stacks/stacks_denovomap.xml [hunk 1]
```diff
@@ -108,28 +110,27 @@ ${input.display_name}::${input}
...
-            <param name="disable_calling" type="boolean" checked="false" default="false" label="disable calling haplotypes from secondary reads" />
+            <param name="m" type="integer" value="-1" label="Minimum number of identical raw reads required to create a stack" help="leave -1 if you don't use the parameter [-m]" />
+            <param name="M" type="integer" value="2" label="Number of mismatches allowed between loci when processing a single individual [-M]"/>
+            <param name="N" type="integer" value="-1" label="Number of mismatches allowed when aligning secondary reads" help="leave -1 if you don't use the parameter [-N]" />
+            <param name="n" type="integer" value="0" label="specify the number of mismatches allowed between loci when building the catalog [-n]"/>
+            <param name="t" type="boolean" checked="false" default="false" label="remove, or break up, highly repetitive RAD-Tags in the ustacks program [-t]" />
+            <param name="H" type="boolean" checked="false" default="false" label="disable calling haplotypes from secondary reads [-H]" />
```
Comment (lecorguille) [inline tools/stacks/stacks_denovomap.xml]: I guess I should use `argument="-m"`
But I wanted to run planemo test before try this solution.

Comment (bgruening) [inline tools/stacks/stacks_denovomap.xml]: Yeah argument='-m' is a handy shortcur, however in your case I'm not sure. Is -m the original argument underlying the stacks tool? If so this is fine. If not -m will appear in the help text indicating the user the underlying argument is -m, but it's not a stacks argument ... so in this case I would prefer not to use arguement='-m'

**Comment by bgruening** [comment]
Thanks!

File: tools/stacks/stacks_denovomap.xml (hunk 1 above)
Comment (lecorguille) [inline tools/stacks/stacks_denovomap.xml]: Yes, they can be mapped on the original arguments: -m, -M, -N, -n, -t, -H,  --alpha, --bound_low and bound_highhttp://catchenlab.life.illinois.edu/stacks/comp/denovo_map.php

Comment (bgruening) [inline tools/stacks/stacks_denovomap.xml]: Awesome! Than go for argument=""

**Pull request merged**